pkg/alertmanager/alertmanager.go

```diff
@@ -224,7 +224,9 @@ func (am *alertmanager) createDiagnosisFromPrometheusAlert(triggers []diagnosisv
 
 				// Create diagnosis according to the prometheus alert.
 				now := time.Now()
-				name := fmt.Sprintf("%s.%s.%s.%d", PrometheusAlertGeneratedDiagnosisPrefix, strings.ToLower(alert.Name()), alert.Fingerprint().String()[:7], now.Unix())
+				// Format alert name to prevent spaces in the diagnosis name.
+				alertName := strings.ReplaceAll(alert.Name(), " ", "-")
```

Comment:
The replacement should not be an arbitrary behavior. It should be under the control of a user.

Reply:
It will bring too much complexity to alert creation, I now change to a more general way to solve this naming problem.